src/main/java/slimeknights/tconstruct/tables/recipe/part/PartRecipe.java

```diff
@@ -0,0 +1,99 @@
+package slimeknights.tconstruct.tables.recipe.part;
+
+import net.minecraft.inventory.IInventory;
+import net.minecraft.item.ItemStack;
+import net.minecraft.item.crafting.IRecipe;
+import net.minecraft.item.crafting.IRecipeSerializer;
+import net.minecraft.item.crafting.IRecipeType;
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.world.World;
+import slimeknights.tconstruct.tables.TinkerTables;
+
+public class PartRecipe implements IRecipe<IInventory> {
```

Comment:
Extend `IInventory` into an interface that better suits the part builder. Should contain methods such as `getInput` (item in material slot), `getMaterial` (computed from the input), and `getPattern` (selected pattern button resource location). Maybe also have a `getPatternItem` for that slot instead of the hardcoded index. Have the interface default anything we don't care about, maybe pulling those defaults to a base `IInventoryHasToManyMethods`